### PR TITLE
Sitl clean 

### PR DESCRIPTION
**Pull request by khancyr**
Rework from https://github.com/ArduPilot/ardupilot/pull/5639 without the cast modification. 

fixing some style issue : catch with cpplint, cppcheck, clion ide, clang-tidy
rework include from
master.pdf
to
clean.pdf : tools used : clion ide, cpplint, https://www.flourish.org/cinclude2dot/ (only include in ap_hal_sitl are show)
correct some variable, param, function
make some variable const

Work done by Skeyetech

### CODE REVIEW

File: libraries/AP_HAL_SITL/GPIO.cpp [hunk 1]
```diff
@@ -63,7 +64,7 @@ bool GPIO::attach_interrupt(uint8_t interrupt_num, AP_HAL::Proc p,
     return true;
 }
 
-bool GPIO::usb_connected(void)
+bool GPIO::usb_connected()
```
Comment (WickedShell) [inline libraries/AP_HAL_SITL/GPIO.cpp]: I'd strongly rather keep the `(void)` expression on function definitions and declaration, and I believe this is the consistent widespread way the code base is written.

Comment (khancyr) [reply libraries/AP_HAL_SITL/GPIO.cpp]: Ok I can keep it, even if I don't find it usefull. And it is more or less consistent in the code base ^^

File: libraries/AP_HAL_SITL/AP_HAL_SITL_Namespace.h [hunk 2]
```diff
@@ -26,4 +26,5 @@ static const uint8_t SITL_RC_INPUT_CHANNELS = 16;
 // Number of rc output channels
 static const uint8_t SITL_RC_OUTPUT_CHANNELS = 16;
 
+static const uint8_t DEFAULT_RCOUTPUT_FREQ = 50;  // PWM RC : 50Hz
```
Comment (OXINARF) [inline libraries/AP_HAL_SITL/AP_HAL_SITL_Namespace.h]: If this is only used in RCOutput, why is this declared here instead of in the RCOutput class?

You can even remove the initialization of `_freq_hz` in the constructor and make the initialization directly in the declaration in the class.

File: libraries/AP_HAL_SITL/SITL_State.h [hunk 3]
```diff
@@ -173,7 +173,7 @@ class HALSITL::SITL_State {
     };
     uint8_t store_index_mag;
     uint32_t last_store_time_mag;
-    VectorN<readings_mag, mag_buffer_length> buffer_mag;
+    VectorN<readings_mag, MAG_BUFFER_LENGTH> buffer_mag;
     uint32_t time_delta_mag;
     uint32_t delayed_time_mag;
```
Comment (OXINARF) [inline libraries/AP_HAL_SITL/SITL_State.h]: Isn't all this mag stuff unused?

File: libraries/AP_HAL_SITL/GPIO.cpp (hunk 1 above)
Comment (OXINARF) [reply libraries/AP_HAL_SITL/GPIO.cpp]: I actually don't like the void, it's completely useless. Also, it's certainly not consistent across the code base. That said, if @WickedShell feels strongly about it, I'd say to leave it.

File: libraries/AP_HAL_SITL/RCInput.h [hunk 4]
```diff
@@ -24,7 +24,7 @@ class HALSITL::RCInput : public AP_HAL::RCInput {
     SITL_State *_sitlState;
 
     /* override state */
-    uint16_t _override[SITL_RC_INPUT_CHANNELS];
+    uint16_t _override[SITL_RC_INPUT_CHANNELS]{};
```
Comment (OXINARF) [inline libraries/AP_HAL_SITL/RCInput.h]: Why is this needed? This should already be 0 on construction.

**Comment by khancyr** [comment]
closed as too old
